sample templates/end_year/SchoolReportTemplate_ZeugnisTemplate_Commands.tex

```diff
@@ -17,34 +15,41 @@
 \newcommand{\gradeDefault}{&  &  &  & \\\hline}
 
 
-\newcommand\levelRow[1]{\phantom{PH} & \multicolumn{1}{m{.07\linewidth}|}{\centering #1} & \multicolumn{4}{c|}{}\\\hline}
-\newcommand{\levelSeven}{\multicolumn{1}{m{.07\linewidth}|}{\centering\tiny bis Klasse 7 ohne Niveau}}
-\newcommand{\levelEight}{\multicolumn{1}{m{.07\linewidth}|}{\centering\tiny bis Klasse 8 ohne Niveau}}
-\newcommand{\levelNine}{\multicolumn{1}{m{.07\linewidth}|}{\centering\tiny bis Klasse 9 ohne Niveau}}
+\newcommand\levelRow[1]{\multicolumn{5}{|c|}{#1}\\\hline}
+\newcommand{\levelOne}{\levelRow{\centering Du hast vorwiegend auf {\color{greenEnglish} Anforderungsebene 1} gearbeitet.}}
+\newcommand{\levelTwo}{\levelRow{\centering Du hast vorwiegend auf {\color{blue} Anforderungsebene 2} gearbeitet.}}
+\newcommand{\levelThree}{\levelRow{\centering Du hast vorwiegend auf {\color{red} Anforderungsebene 3} gearbeitet.}}
+\newcommand{\levelSeven}{\levelRow{\centering bis Klasse 7 ohne Anforderungsebene}}
+\newcommand{\levelEight}{\levelRow{\centering bis Klasse 8 ohne Anforderungsebene}}
+\newcommand{\levelNine}{\levelRow{\centering bis Klasse 9 ohne Anforderungsebene}}
+\newcommand{\noLevel}{}
+
 
-%\setlength{\extrarowheight}{0pt}
-\newcommand\competencyMajorSubject[2]{\makecell[{{>{\parindent0em}p{\linewidth}}}]{#1} & #2}
-\newcommand\competencyMinorSubject[3]{\makecell[{{{>{\parindent0em}p\linewidth}}}]{#1} & #3 #2}
+% #1 ... competence
+% #2 ... grade
+\newcommand\competencyMajorSubject[2]{\makecell[{{>{\parindent0em}p{\linewidth}}}]{#1} #2}
 
+% #1 ... subject
+% #2 ... competencies
+% #3 ... level
 \newcommand\competencyTableMajorSubject[3]{
 	\renewcommand{\arraystretch}{1.5}
-	\begin{tabularx}{\linewidth}{|K{X}|C{.07\linewidth}|C{\tableGradeWidth}|C{\tableGradeWidth}|C{\tableGradeWidth}|C{\tableGradeWidth}|}
+	\begin{tabularx}{\linewidth}{|K{X}|C{\tableGradeWidth}|C{\tableGradeWidth}|C{\tableGradeWidth}|C{\tableGradeWidth}|}
 		\hline
-		\multicolumn{6}{|>{\columncolor{egelightblue}}l|}{\textit{\textbf{#1}}} \\
-		\hline
-		#2
-		\levelRow{#3}
-	\end{tabularx}\vspace{2pt}
-}
-\newcommand\competencytable[2]{
-	\begin{tabularx}{\linewidth}{|K{X}|C{.07\linewidth}|C{\tableGradeWidth}|C{\tableGradeWidth}|C{\tableGradeWidth}|C{\tableGradeWidth}|}
-		\hline
-		\multicolumn{6}{|>{\columncolor{egelightblue}}l|}{\textit{\textbf{#1}}} \\
+		\multicolumn{5}{|>{\columncolor{egelightblue}}l|}{\textit{\textbf{#1}}} \\
 		\hline
 		#2
+		#3
 	\end{tabularx}\vspace{2pt}
 }
 
+% argument is the page threshold until a newline should be added
+\newcommand\newOptionalNewPage[1]{
+	\ifnum #1>\value{page}
+		\newpage
+		\phantom{~}
+	\fi
+}
```

Comment:
Nice! Micha spent quite some time to figure this out - with this, a report can always be 12 pages (with filler pages in between).